src/win32/thread.c

```diff
@@ -240,14 +215,26 @@ int pthread_rwlock_wrunlock(pthread_rwlock_t *lock)
 	return 0;
 }
 
-int pthread_rwlock_destroy(pthread_rwlock_t *lock)
+int git_rwlock_free(git_rwlock *lock)
 {
 	if (!win32_srwlock_initialize)
 		DeleteCriticalSection(&lock->native.csec);
 	git__memzero(lock, sizeof(*lock));
 	return 0;
 }
 
+int git_num_processors(void)
```

Comment:
Aha, yes, it seems like we now have [`git_online_cpus`](https://github.com/libgit2/libgit2/blob/77394a27af283b366fa8bb444d29670131bfa104/src/thread-utils.c#L32) and so this is probably unused.  Let's definitely get rid of this one.  I'll let you use your discretion on the others.